modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java

```diff
@@ -831,6 +849,108 @@ public void testSendNotificationWithSystemObjectDefinition()
 			notificationQueueEntries.get(0));
 
 		objectActionLocalService.deleteObjectAction(objectAction);
+
+		// On Commerce Order payment status update
+
+		ObjectDefinition commerceOrderObjectDefinition =
+			_objectDefinitionLocalService.fetchObjectDefinitionByClassName(
+				TestPropsValues.getCompanyId(), CommerceOrder.class.getName());
+
+		Set<String> objectActionTriggerKeys = new HashSet<>();
+
+		ListUtil.isNotEmptyForEach(
+			_objectActionTriggerRegistry.getObjectActionTriggers(
+				commerceOrderObjectDefinition.getClassName()),
+			objectActionTrigger -> objectActionTriggerKeys.add(
+				objectActionTrigger.getKey()));
+
+		Assert.assertTrue(
+			objectActionTriggerKeys.contains(
+				DestinationNames.COMMERCE_ORDER_STATUS));
+
+		Assert.assertTrue(
+			objectActionTriggerKeys.contains(
+				DestinationNames.COMMERCE_PAYMENT_STATUS));
+
+		NotificationTemplate commerceOrderNotificationTemplate =
+			notificationTemplateLocalService.addNotificationTemplate(
+				NotificationTemplateUtil.createNotificationContext(
+					TestPropsValues.getUser(),
+					commerceOrderObjectDefinition.getObjectDefinitionId(),
+					RandomTestUtil.randomString(),
+					RandomTestUtil.randomString(),
+					NotificationTemplateConstants.EDITOR_TYPE_RICH_TEXT,
+					Arrays.asList(
+						createNotificationRecipientSetting(
+							"from", "[%CURRENT_USER_EMAIL_ADDRESS%]"),
+						createNotificationRecipientSetting(
+							"fromName",
+							Collections.singletonMap(
+								LocaleUtil.US, "[%CURRENT_USER_FIRST_NAME%]")),
+						createNotificationRecipientSetting(
+							"to", "[%CURRENT_USER_EMAIL_ADDRESS%]")),
+					RandomTestUtil.randomString(),
+					NotificationConstants.TYPE_EMAIL, Collections.emptyList()));
+
+		ObjectAction commerceOrderObjectAction =
+			_objectActionLocalService.addObjectAction(
+				RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+				commerceOrderObjectDefinition.getObjectDefinitionId(), true,
+				StringPool.BLANK, RandomTestUtil.randomString(),
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				RandomTestUtil.randomString(),
+				ObjectActionExecutorConstants.KEY_NOTIFICATION,
+				DestinationNames.COMMERCE_PAYMENT_STATUS,
+				UnicodePropertiesBuilder.put(
+					"notificationTemplateId",
+					commerceOrderNotificationTemplate.
+						getNotificationTemplateId()
+				).build(),
+				false);
```

Comment:
@icgsbr please create a private method to reuse code:

```
private ObjectAction _addObjectAction(
		String objectActionTriggerKey, ObjectDefinition objectDefinition,
		String to)
	throws Exception {

	NotificationTemplate notificationTemplate =
		notificationTemplateLocalService.addNotificationTemplate(
			NotificationTemplateUtil.createNotificationContext(
				TestPropsValues.getUser(), 0, RandomTestUtil.randomString(),
				RandomTestUtil.randomString(),
				NotificationTemplateConstants.EDITOR_TYPE_RICH_TEXT,
				Arrays.asList(
					createNotificationRecipientSetting(
						"bcc", "[%CURRENT_USER_EMAIL_ADDRESS%]"),
					createNotificationRecipientSetting(
						"cc",
						"[%CURRENT_USER_EMAIL_ADDRESS%],cc@liferay.com"),
					createNotificationRecipientSetting(
						"from", "[%CURRENT_USER_EMAIL_ADDRESS%]"),
					createNotificationRecipientSetting(
						"fromName",
						Collections.singletonMap(
							LocaleUtil.US, "[%CURRENT_USER_FIRST_NAME%]")),
					createNotificationRecipientSetting("to", to)),
				RandomTestUtil.randomString(),
				NotificationConstants.TYPE_EMAIL, Collections.emptyList()));

	return _objectActionLocalService.addObjectAction(
		RandomTestUtil.randomString(), TestPropsValues.getUserId(),
		objectDefinition.getObjectDefinitionId(), true, StringPool.BLANK,
		RandomTestUtil.randomString(),
		LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
		LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
		RandomTestUtil.randomString(),
		ObjectActionExecutorConstants.KEY_NOTIFICATION,
		objectActionTriggerKey,
		UnicodePropertiesBuilder.put(
			"notificationTemplateId",
			notificationTemplate.getNotificationTemplateId()
		).build(),
		false);
}
```

usages:

```
// On Account Entry update

ObjectDefinition accountEntryObjectDefinition =
	objectDefinitionLocalService.fetchObjectDefinition(
		TestPropsValues.getCompanyId(),
		AccountEntry.class.getSimpleName());

ObjectAction objectAction = _addObjectAction(
	ObjectActionTriggerConstants.KEY_ON_AFTER_UPDATE,
	accountEntryObjectDefinition,
	"[%ACCOUNTENTRY_AUTHOR_EMAIL_ADDRESS%]");
```

and:

```
objectAction = _addObjectAction(
	DestinationNames.COMMERCE_PAYMENT_STATUS,
	commerceOrderObjectDefinition, "[%CURRENT_USER_EMAIL_ADDRESS%]");
```